src/libstd/rt/unwind/seh64_gnu.rs

```diff
@@ -0,0 +1,231 @@
+// 2. That the first parameter of the exception is a pointer to an extra data area (PanicData).
+// Since these assumptions do not generally hold true for foreign exceptions (system faults,
+// C++ exceptions, etc), we make no attempt to invoke our landing pads (and, thus, destructors!)
+// for anything other than RUST_PANICs. This is considered acceptable, because the behavior of
+// throwing exceptions through a C ABI boundary is undefined.
+
+#[no_mangle] // referenced from rust_try.ll
+#[cfg(not(test))]
+pub unsafe extern "C" fn rust_eh_personality_catch(
+    exceptionRecord: *mut EXCEPTION_RECORD,
+    establisherFrame: LPVOID,
+    contextRecord: *mut CONTEXT,
+    dispatcherContext: *mut DISPATCHER_CONTEXT
+) -> EXCEPTION_DISPOSITION
+{
+    let er = &*exceptionRecord;
+    let dc = &*dispatcherContext;
+    rtdebug!("rust_eh_personality_catch: code={:X}, flags={:X}, frame={:X}, ip={:X}",
+        er.ExceptionCode, er.ExceptionFlags, establisherFrame as usize, dc.ControlPc as usize);
+
+    if er.ExceptionFlags & EXCEPTION_UNWIND == 0 { // we are in the dispatch phase
+        if er.ExceptionCode == RUST_PANIC {
+            if let Some(lpad) = find_landing_pad(dc) {
+                RtlUnwindEx(establisherFrame,
+                            lpad as LPVOID,
+                            exceptionRecord,
+                            er.ExceptionInformation[0] as LPVOID, // pointer to PanicData
+                            contextRecord,
+                            dc.HistoryTable);
+                rtabort!("could not unwind");
+            }
+            rtabort!("could not locate the landing pad");
+        }
+    }
+    ExceptionContinueSearch
+}
+
+#[lang="eh_personality"]
+#[cfg(not(test))]
+unsafe extern "C" fn rust_eh_personality(
+    exceptionRecord: *mut EXCEPTION_RECORD,
+    establisherFrame: LPVOID,
+    contextRecord: *mut CONTEXT,
+    dispatcherContext: *mut DISPATCHER_CONTEXT
+) -> EXCEPTION_DISPOSITION
+{
+    let er = &*exceptionRecord;
+    let dc = &*dispatcherContext;
+    rtdebug!("rust_eh_personality: code={:X}, flags={:X}, frame={:X}, ip={:X}",
+        er.ExceptionCode, er.ExceptionFlags, establisherFrame as usize, dc.ControlPc as usize);
+
+    if er.ExceptionFlags & EXCEPTION_UNWIND == 0 {      // we are in the dispatch phase
+        if er.ExceptionCode == RUST_PANIC {
+            if let Some(lpad) = find_landing_pad(dc) {
+                rtdebug!("unwinding to landing pad {:X}", lpad);
+
+                RtlUnwindEx(establisherFrame,
+                            lpad as LPVOID,
+                            exceptionRecord,
+                            er.ExceptionInformation[0] as LPVOID, // pointer to PanicData
+                            contextRecord,
+                            dc.HistoryTable);
+                rtabort!("could not unwind");
+            }
+        }
+    }
+    ExceptionContinueSearch
+}
+
+#[no_mangle] // referenced from landing pads
```

Comment:
Can you expand this to a comment explaining what this function is doing? It'd be nice to mention that this is called from all landing pads and is injected by LLVM, and it's also why it's `pub` + `#[no_mangle]`.

Also, can you remove the `"C"` part from the header?